app/containers/ItunesContainer/reducer.js

```diff
@@ -0,0 +1,43 @@
+/*
+ *
+ * ItunesContainer reducer
+ *
+ */
+import produce from 'immer';
+import { createActions } from 'reduxsauce';
+import get from 'lodash/get';
+
+export const initialState = {
+  searchTerm: null,
+  gridData: [],
+  searchesError: null
+};
+
+export const { Types: itunesContainerTypes, Creators: itunesContainerCreators } = createActions({
+  searchItunes: ['searchTerm'],
+  successSearchItunes: ['data'],
+  failureSearchItunes: ['error'],
+  clearGridData: []
+});
+
+/* eslint-disable default-case, no-param-reassign */
+export const itunesContainerReducer = (state = initialState, action) =>
+  produce(state, (draft) => {
+    switch (action.type) {
+      case itunesContainerTypes.SEARCH_ITUNES:
+        draft.searchTerm = action.searchTerm;
+        break;
+      case itunesContainerTypes.SUCCESS_SEARCH_ITUNES:
+        draft.searchesError = null;
+        draft.gridData = action.data;
+        break;
+      case itunesContainerTypes.FAILURE_SEARCH_ITUNES:
+        draft.gridData = [];
+        draft.searchesError = get(action.error, 'messages', 'something_went_wrong');
+        break;
+      default:
```

Comment:
with immer we don't need to handle default case

Reply:
Noted, thank you for letting me know

Reply:
Updated in the PR